### DISCUSSION TITLE
Scene Graph (Parent / Children) component

### DISCUSSION THREAD

**Discussion by oaklandgit** [replit/kaboom#176] [Ideas]
Hi,

I posted this as a feature request in Issues, but I think it might be better for a discussion here, as I am trying to build this component myself, but am hitting some snags.

I wanted a "parent/child" component (e.g. a turret as a child of a tank in this example) so I did my best to make one here:
https://KaboomParentChild.larrystone.repl.co

But I recognize it has a few issues for a full-fledged component:

1) In order to affect children (e.g. their rotation or position) you need to target the custom "localX, localY, and localAngle" variables, which seems wrong. Ideally, the child should be positionable via the pos() component, and the "children()" component should mange the children accordingly.

2) A proper scene graph component should also be able to manage the origin offset, and alpha transparency of the children.

I will continue to try to improve this, but any advice would be appreciated!

Also, are there others out there who would find a children() component to be useful?

**Comment by slmjkdbtl** [comment]
Yes I don't think it's possible to do it with external APIs. I'm interested in doing it but it does require some non-trivial refactor, will get to it when I got more time.

**Reply by oaklandgit** [reply]
OK thanks. I may still continue trying to hack it with the existing APIs. It's amazing what you've done already with this project!
